cache/src/main/scala/coursier/Cache.scala

```diff
@@ -1123,6 +1155,8 @@ object Cache {
 
     def gettingLength(url: String): Unit = {}
     def gettingLengthResult(url: String, length: Option[Long]): Unit = {}
+
+    def removedCorruptFile(url: String, file: File, reason: Option[String]): Unit
```

Comment:
Let's keep the full original `FileError` in `reason` instead of a `String`.

Reply:
done